fence/README.md

```diff
@@ -0,0 +1,153 @@
+| -------- | ------------------------------------------------------------------------------------- |
+| CO       | Container Orchestration system that communicates with plugins using CSI service RPCs. |
+| SP       | Storage Provider, the vendor of a CSI plugin implementation.                          |
+| DR       | Disaster Recovery.                                                                    |
+| RPC      | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).         |
+
+## Objective
+
+Define a standard that will enable storage vendors (SP) to
+performing cluster level fencing on required addresses.
+
+### Goals in MVP
+
+The new extension will define a procedure that allows SP to fence I/O from desired nodes.
+
+### Non-Goals in MVP
+
+* Application level fencing.
+
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and
+packaging recommendations for a storage provider (SP) to implement
+fencing operation. The interface declares the RPCs that a plugin
+MUST expose.
+
+## RPC Interface
+
+* **Controller Service**: The Controller plugin MUST implement these sets of
+  RPCs.
+
+```protobuf
+syntax = "proto3";
+package fence;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = ".;fence";
+
+extend google.protobuf.FieldOptions {
+  // Indicates that a field MAY contain information that is sensitive
+  // and MUST be treated as such (e.g. not logged).
+  bool fence_secret = 1099;
+}
+
+// Controller holds the RPC method for performing fencing operations.
+service Controller {
+  // EnableClusterFence RPC call to perform a fencing operation.
+  rpc EnableClusterFence (EnableClusterFenceRequest)
+  returns (EnableClusterFenceResponse) {}
+
+  // DisableClusterFence RPC call to remove a list of CIDR blocks from the
+  // list of blocklisted/fenced clients.
+  rpc DisableClusterFence (DisableClusterFenceRequest)
+  returns (DisableClusterFenceResponse) {}
+
+  // ListClusterFence RPC call to provide a list of blocklisted/fenced clients.
+  rpc ListClusterFence(ListClusterFenceRequest)
+  returns (ListClusterFenceResponse){}
+}
+```
+
+### EnableClusterFence
+
+```protobuf
+// EnableClusterFenceRequest contains the information needed to identify
+// the cluster so that the appropriate fencing operation can be
+// performed.
+message EnableClusterFenceRequest {
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 1;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 2 [(fence_secret) = true];
+  // list of CIDR blocks on which the fencing operation is expected to be
+  // performed.
+  repeated CIDRList cidr_list = 3;
+}
+
+// EnableClusterFenceResponse holds the information about the result of the
+// EnableClusterFenceRequest call.
+message EnableClusterFenceResponse {
+    // The message returned on execution as received from SP.
+    string result = 1;
```

Comment:
What kind of result should this be? If it is just a message formatted as a string, without any defined value that can be interpreted by the caller, maybe name `result` differently? gRPC includes a response and error on requests, so be explicit in what you expect to see as attributes in a response.